tracee/event_monitor_ebpf.c

```diff
@@ -891,57 +906,69 @@ static __always_inline int save_args_to_submit_buf(u64 types)
     return 0;
 }
 
+static __always_inline int trace_ret_generic(struct pt_regs *ctx, u32 id, u64 types)
+{
+    context_t context = {};
+
+    if (init_context(&context) || init_submit_buf())
+        return 0;
+
+    context.eventid = id;
+    context.argnum = get_encoded_arg_num(types);
+    context.retval = PT_REGS_RC(ctx);
+    save_to_submit_buf((void*)&context, sizeof(context_t), NONE_T);
+    save_args_to_submit_buf(types);
+
+    events_perf_submit(ctx);
+    return 0;
+}
+
+static __always_inline int trace_ret_generic_fork(struct pt_regs *ctx, u32 id, u64 types)
+{
+    context_t context = {};
+
+    if (init_context(&context) || init_submit_buf())
+        return 0;
+
+    if (!container_mode()) {
+        u32 pid = PT_REGS_RC(ctx);
+        add_pid_fork(pid);
+    }
+
+    context.eventid = id;
+    context.argnum = get_encoded_arg_num(types);
+    context.retval = PT_REGS_RC(ctx);
+    save_to_submit_buf((void*)&context, sizeof(context_t), NONE_T);
+    save_args_to_submit_buf(types);
+
+    events_perf_submit(ctx);
+    return 0;
+}
+
```

Comment:
would be nice to think of a way to merge those almost identical functions. this can be in a future pr